tensorflow/python/ops/nn_ops.py

```diff
@@ -4342,6 +4343,10 @@ def dropout_v2(x, rate, noise_shape=None, seed=None, name=None):
         return x
 
     noise_shape = _get_noise_shape(x, noise_shape)
+
+    if build_info.is_rocm_build:
```

Comment:
Desirably, this branch on the following two conditions:

- The build of TF is a ROCm build
- The operator is placed on GPU

However, I haven't seen one instance in the TF code base that one operator will query if the user system have a GPU on it. The logic is usually the other way around: TF executor assigns operator to the device. Then in the runtime, operator query if a specific kernel implementation is on that device. Because of this, the CPU execution will fail even though we have a python implementation right here.

Reply:
I cannot say what can be upstream but at least with checks you mentioned you can validate the MIOpen dropout implementation to the existing numpy tensorflow implementation.